virtualizarr/tests/test_readers/test_dmrpp.py

```diff
@@ -354,3 +372,38 @@ def test_parse_chunks(
         lengths=expected_lengths, offsets=expected_offsets, paths=expected_paths
     )
     assert result == expected
+
+
+class TestRelativePaths:
+    def test_absolute_path_to_dmrpp_file_containing_relative_path(
+        self,
+        basic_dmrpp_temp_filepath: Path,
+    ):
+        vds = open_virtual_dataset(
+            str(basic_dmrpp_temp_filepath), indexes={}, filetype="dmrpp"
+        )
+        path = vds["x"].data.manifest["0"]["path"]
+
+        # by convention, if dmrpp file path is {PATH}.nc.dmrpp, the data filepath should be {PATH}.nc
+        # and the manifest should only contain absolute file URIs
+        expected_datafile_path_uri = basic_dmrpp_temp_filepath.as_uri().removesuffix(
+            ".dmrpp"
+        )
+        assert path == expected_datafile_path_uri
```

Comment:
@ayushnag currently I think the comments in this test are lying about what it actually tests. I had to change relative to absolute paths in the dmrpp strings above to get all the other tests to pass, but I don't know if that's valid dmrpp, and it means that this test is not testing using a dmrpp containing a relative datafilepath :/

Reply:
I'm not sure why modifying the XML is needed since that "name" attribute is only used if you don't provide a datapath. However the DMRPPVirtualBackend [always provides a datapath](https://github.com/zarr-developers/VirtualiZarr/blob/515262493521a3d0a0659d8290ae28e0797faf47/virtualizarr/readers/dmrpp.py#L50) which is just the filepath the user passed into `open_virtual_dataset`. Basically the input to `open_virtual_dataset` minus `.dmrpp` should be the path that ends up in the manifest.

Reply:
> However the `DMRPPVirtualBackend` [always provides a datapath](https://github.com/zarr-developers/VirtualiZarr/blob/515262493521a3d0a0659d8290ae28e0797faf47/virtualizarr/readers/dmrpp.py#L50) which is just the filepath the user passed into open_virtual_dataset

Right, but most of the tests in `test_dmrpp.py` don't actually use the `DMRPPVirtualBackend` class, they go straight to the `DMRParser`, so the `datapath` doesn't get provided (because its not even known at fixture-creation time). This is what I was originally trying to point out about the structure of these tests.